server/routes/posts.js

```diff
@@ -38,30 +44,107 @@ module.exports = (() => {
       })
     })
 
-    router.post('/create', (req, res) => {
-      
+
+
+    /* Endpoint to provide partial list of posts based on keyword search */
+    router.get('/list/:keyword', (req,res) => {
+      console.log("Endpoint: Listing Posts Based on Keyword Match ")
+     Post.find({body: {$regex: req.params.keyword}}, function(err, posts) {
+        if (err) throw err;
+        if (!posts) {
```

Comment:
Why do we need to create a placeholder post if there is no post matching the keyword?